pkg/controller/propagator/propagation.go

```diff
@@ -676,22 +685,35 @@ func (r *ReconcilePolicy) processTemplates(replicatedPlc *policiesv1.Policy, dec
 			reqLogger.Error(tplErr, "Failed to resolve templates")
 
 			r.recorder.Event(rootPlc, "Warning", "PolicyPropagation",
-				fmt.Sprintf("Failed to resolve templates for policy %s/%s for cluster %s/%s .", rootPlc.GetName(), rootPlc.GetNamespace(), decision.ClusterNamespace, decision.ClusterName))
+				fmt.Sprintf("Failed to resolve templates for cluster %s/%s: %s", decision.ClusterNamespace, decision.ClusterName, tplErr.Error()))
+			//Set an annotation on the policyTemplate(e.g. ConfigurationPolicy)  to the template processing error msg
+			//managed clusters will use this when creating a violation
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+			jsonErr := json.Unmarshal(policyT.ObjectDefinition.Raw, policyTObjectUnstructured)
+			if jsonErr != nil {
+				//it shouldnt get here but if it did just log a msg
+				//its alright, a generic msg will be used on the managedcluster
+				reqLogger.Error(jsonErr, fmt.Sprintf("Error unmarshalling to json for Policy %s, Cluster %s.", rootPlc.GetName(), decision.ClusterName))
+				policyTAnnotations := policyTObjectUnstructured.GetAnnotations()
+				if policyTAnnotations == nil {
+					policyTAnnotations = make(map[string]string)
+				}
+				policyTAnnotations["policy.open-cluster-management.io/hub-templates-error"] = tplErr.Error()
```

Comment:
So the annotation error message only ever gets set if the object definition is invalid JSON?

Could you please explain why we wouldn't want to do this in the event that template was invalid but it's still valid JSON?

Reply:
if you look in the first commit , there is else there , i am sure what happened there

Reply:
Okay, let me know when you get that fixed.

Reply:
Looks like I accidentally removed the adjacent else clause while fixing the format of  json error msg

Reply:
fixed it